metal/resource_metal_device.go

```diff
@@ -678,3 +776,25 @@ func resourceMetalDeviceDelete(d *schema.ResourceData, meta interface{}) error {
 	}
 	return nil
 }
+
+func waitForActiveDevice(d *schema.ResourceData, meta interface{}) error {
+	// Wait for the device so we can get the networking attributes that show up after a while.
+	state, err := waitForDeviceAttribute(d, []string{"active", "failed"}, []string{"queued", "provisioning", "reinstalling"}, "state", meta)
+	if err != nil {
+		d.SetId("")
+		fErr := friendlyError(err)
+		if isForbidden(fErr) {
```

Comment:
@t0mk We may want to rewrite this as a separate issue. I believe 404 is returned now for devices that have been moved to the failed provisioning project.

We can also offer more details than `the Equinix Metal team will investigate`, which may be misleading.

I'm thinking:
- return the error as-is wrapped with error text that we timed out waiting for active state
- include the instance id in the error
- if the error was 403/404, log that the Equinix Metal team will investigate (including a link to where this is stated in the docs) (don't say 'provisioning time limit exceeded', it could be sometime else, like invalid ipxe script, etc)

Reply:
Created https://github.com/equinix/terraform-provider-metal/issues/153